plugin-server/src/main/ingestion-queues/batch-processing/each-batch-onevent.ts

```diff
@@ -18,8 +18,14 @@ export async function eachMessageAppsOnEventHandlers(
 ): Promise<void> {
     const clickHouseEvent = JSON.parse(message.value!.toString()) as RawClickHouseEvent
 
-    if (queue.pluginsServer.pluginConfigsPerTeam.has(clickHouseEvent.team_id)) {
-        const event = convertToIngestionEvent(clickHouseEvent)
+    const pluginConfigs = queue.pluginsServer.pluginConfigsPerTeam.get(clickHouseEvent.team_id)
+    if (pluginConfigs) {
+        // Elements parsing can be extremely slow, so we skip it for some plugins
+        const skipElementsChain = pluginConfigs.every((pluginConfig) =>
+            process.env.SKIP_ELEMENTS_PARSING_PLUGINS?.split(',').includes(pluginConfig.plugin_id.toString())
```

Comment:
`eachMessageAppsOnEventHandlers` is in the hot path (executed for each message), so we should not have a string array allocation + a linear search in the array here.

Let's:
  - add a proper field to `PluginsServerConfig`, instead of using rogue envvrs
  - parse the string once in a `HashSet<number>`, and have the hot path only check for presence in the hashset.

We use this pattern a lot, so I think it's worth adding a facility in `config.ts` that we'll be able to reuse.